lib/semian/net_http.rb

```diff
@@ -87,14 +88,24 @@ def disabled?
       raw_semian_options.nil?
     end
 
+    def fatal_server_errors?
+      raw_semian_options.fetch(:fatal_server_errors)
+    end
+
     def connect
       return raw_connect if disabled?
       acquire_semian_resource(adapter: :http, scope: :connection) { raw_connect }
     end
 
     def request(req, body = nil, &block)
       return raw_request(req, body, &block) if disabled?
-      acquire_semian_resource(adapter: :http, scope: :query) { raw_request(req, body, &block) }
+      acquire_semian_resource(adapter: :http, scope: :query) do
+        result = raw_request(req, body, &block)
+
+        result.error! if result.is_a?(::Net::HTTPServerError) && fatal_server_errors?
```

Comment:
@sirupsen What do you think about the approach in the last commit? We would be now able to trip the circuit on 5xxs by simply marking them as failures given the user sets that option to `true`, and that logic is now decoupled from the adapters.